src/libraries/Microsoft.PowerFx.Core/Localization/StringResources.cs

```diff
@@ -230,18 +233,26 @@ bool IExternalStringResources.TryGetErrorResource(ErrorResourceKey resourceKey,
             // Error resources are a bit odd and need to be reassembled from separate keys.
             // Check to see if we've already retrieved one for this locale/key combo before 
             // rebuilding the full error resource.
-
-            if (_errorResources.TryGetValue(locale, out var localizedErrorResources))
+            try
             {
-                if (localizedErrorResources.TryGetValue(resourceKey.Key, out resourceValue))
+                _lock.EnterReadLock();
+                if (_errorResources.TryGetValue(locale, out var localizedErrorResources))
```

Comment:
With the previous change, the lock around here was removed causing this to no longer be under the lock.  But we can't access a non-thread-safe collection in a multi-threaded way, even if it's a read. 

Use a reader-writer lock to minimize contention.